plugins/Annotations/API.php

```diff
@@ -188,17 +216,27 @@ public function get(int $idSite, int $idNote): array
      *                 ),
      *                 8 => array(...)
      *               )
+     * @throws Exception when user is not a superuser
      */
     public function getAll(string $idSite, ?string $date = null, string $period = 'day', ?int $lastN = null): array
     {
         Piwik::checkUserHasViewAccess($idSite);
 
-        $annotations = new AnnotationList($idSite);
-
-        // if date/period are supplied, determine start/end date for search
-        list($startDate, $endDate) = Annotations::getDateRangeForPeriod($date ?? false, $period, $lastN ?? false);
+        $ids = array_map('intval', Site::getIdSitesFromIdSitesString($idSite));
+        $model = new Model();
+        $annotations = [];
+        foreach ($ids as $id) {
+            [$startDate, $endDate] = Annotations::getDateRangeForPeriod($date, $period, $lastN ?? false, $id);
```

Comment:
```suggestion
            [$startDate, $endDate] = Annotations::getDateRangeForPeriod($date ?? false, $period, $lastN ?? false, $id);
```

If the date is not passed to the API it will break things.

Interesting to see there is no test (yet) for that usage :thinking:

Reply:
Update to work when no date is provided and added a test for that.